src/main/java/mate/academy/MergeSortAction.java

```diff
@@ -1,8 +1,61 @@
 package mate.academy;
 
-public class MergeSortAction {
+import java.util.Arrays;
+import java.util.concurrent.RecursiveAction;
+
+public class MergeSortAction extends RecursiveAction {
+    private static final int THRESHOLD = 2;
+
+    private final int[] array;
+    private final int start;
+    private final int end;
 
     public MergeSortAction(int[] array) {
+        this(array, 0, array.length);
+    }
+
+    private MergeSortAction(int[] array, int start, int end) {
+        this.array = array;
+        this.start = start;
+        this.end = end;
```

Comment:
In `merge sort`, it's standard practice to pass only the subarray as a recursive parameter. This approach makes the `start` and `end` parameters unnecessary, reducing the amount of state maintained within the object - minimizing states within the object is beneficial.